gettsim/transfers/rente.py

```diff
@@ -274,25 +300,419 @@ def ges_rente_zugangsfaktor(
+
+    Parameters
+    ----------
+    pflichtbeitragszeit
+        See basic input variable :ref:`pflichtbeitragszeit <pflichtbeitragszeit>`.
+    freiw_beitragszeit
+        See basic input variable :ref:`freiw_beitragszeit <freiw_beitragszeit>`.
+    ersatzzeit
+        See basic input variable :ref:`ersatzzeit <ersatzzeit>`.
+
+    Returns
+    -------
+    wartezeit in years
+    """
+    out = (pflichtbeitragszeit + freiw_beitragszeit + ersatzzeit) / 12
+
+    return out
+
+
+# todo imaybe implement default  for zeiten if vars
+#  not available? somewhere swtich for eligibility in interface?
+
+
+def ges_rente_wartezeit_15(
+    pflichtbeitragszeit: float, freiw_beitragszeit: float, ersatzzeit: float
+) -> float:
+    """Aggregates time periods that are relevant for the pension
+    for women and Leistungen zur Teilhabe. Wartezeit von 15 Jahren.
+
+    Parameters
+    ----------
+    pflichtbeitragszeit
+        See basic input variable :ref:`pflichtbeitragszeit <pflichtbeitragszeit>`.
+    freiw_beitragszeit
+        See basic input variable :ref:`freiw_beitragszeit <freiw_beitragszeit>`.
+    ersatzzeit
+        See basic input variable :ref:`ersatzzeit <ersatzzeit>`.
 
     Returns
     -------
+    wartezeit in years
     """
-    out = piecewise_polynomial(
-        x=geburtsjahr,
-        thresholds=ges_rente_params["regelaltersgrenze"]["thresholds"],
-        rates=ges_rente_params["regelaltersgrenze"]["rates"],
-        intercepts_at_lower_thresholds=ges_rente_params["regelaltersgrenze"][
-            "intercepts_at_lower_thresholds"
-        ],
-    )
+    out = (pflichtbeitragszeit + freiw_beitragszeit + ersatzzeit) / 12
+
+    return out
+
+
+def ges_rente_wartezeit_35(
+    rententrechtl_zeit: float,
+    pflichtbeitragszeit: float,
+    freiw_beitragszeit: float,
+    anrechnungszeit: float,
+    ersatzzeit: float,
+    kinder_berückz: float,
+    pflege9295_berückz: float,
+) -> float:
+    """Aggregates time periods that are relevant for the eligibility of pension
+    for long-term insured. Wartezeit von 35 Jahren. All "rentenrechtliche Zeiten"
+    are considered.
+
+    Parameters
+    ----------
+    rententrechtl_zeit
+        See basic input variable :ref:`rententrechtl_zeit <rententrechtl_zeit>`.
+    pflichtbeitragszeit
+        See basic input variable :ref:`pflichtbeitragszeit <pflichtbeitragszeit>`.
+    freiw_beitragszeit
+        See basic input variable :ref:`freiw_beitragszeit <freiw_beitragszeit>`.
+    ersatzzeit
+        See basic input variable :ref:`ersatzzeit <ersatzzeit>`.
+    anrechnungszeit
+        See basic input variable :ref:`anrechnungszeit <anrechnungszeit>`.
+    kinder_berückz
+        See basic input variable :ref:`kinder_berückz <kinder_berückz>`.
+    pflege9295_berückz
+        See basic input variable :ref:`pflege9295_berückz <pflege9295_berückz>`.
+
+    Returns
+    -------
+    wartezeit in years
+
+    """
+    if rententrechtl_zeit > 0:  # is not missing
```

Comment:
If I understand correctly, it is defined as if ``rentenrechtl_zeit`` can be an input variable.

We should rather only use the disagregated variables as "basic" inputs. The variable `ges_rente_wartezeit_35` can always be overridden ny the user if rentenrechtl_zeit is available.